app/src/main/java/com/duckduckgo/app/browser/serviceworker/BrowserServiceWorkerClient.kt

```diff
@@ -22,10 +22,16 @@ import androidx.webkit.ServiceWorkerClientCompat
 import com.duckduckgo.app.browser.RequestInterceptor
 import com.duckduckgo.app.global.exception.UncaughtExceptionRepository
 import com.duckduckgo.app.global.exception.UncaughtExceptionSource
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
 import kotlinx.coroutines.runBlocking
 import timber.log.Timber
+import javax.inject.Inject
 
-class BrowserServiceWorkerClient(
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class BrowserServiceWorkerClient @Inject constructor(
```

Comment:
I would mark this as `internal` for two reasons:
* nit about correctness as you're now accessing it manually
* it ensures this is not contributed to DI, because Dagger will fail the build as it requires contributed classes to be public

```suggestion
internal class BrowserServiceWorkerClient @Inject constructor(
```